searchengine.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import os
+
+from cse.SearchEngine import SearchEngine
+from cse.__init__ import __version__
+from cse.__init__ import __title__
+
+# This path is relativ to this python script
+DIRECTORY = os.path.join(".")
+# please specify the filenames of the raw comments data
+COMMENTSFILE = "comments.csv"
+AUTHORFILE = "authors.csv"
+ARTICLEFILE = "articles.csv"
+
+
+def checkIndexFilenames(filenameSource):
+    commentsFilename = filenameSource.split(":")[1].strip()
+
+    # check files
+    for filename in [commentsFilename, AUTHORFILE, ARTICLEFILE]:
+        if not os.path.exists(os.path.join(DIRECTORY, filename)):
+            raise ValueError("File {} relative to this scripts directory could not be found!".format(os.path.join(DIRECTORY, filename)))
+
+    return commentsFilename, AUTHORFILE, ARTICLEFILE
+
+
+def read_queries(filename):
+    if not os.path.exists(filename):
+        raise ValueError("File {} does not exist!".format(filename))
+    queries = []
+    with open(filename, 'rt', encoding="UTF-8") as file:
+        for line in file:
+            queries.append(line)
+    return queries
+
+
+def print_results_to(results, filename, idsOnly=False):
+    with open(filename, 'wt', encoding="UTF-8") as file:
+        for res in results:
+            if idsOnly:
+                file.write(str(res) + "\n")
+            else:
+                file.write("{}, {}".format(res[0], res[1].replace("\n", "\\n")) + "\n")
+
+
+def main():
+    # parse args
+    parser = argparse.ArgumentParser(description='Search Engine for comments on news websites')
+    parser.add_argument("query", help="a txt file with one boolean, keyword, phrase, ReplyTo, or Index query per line")
+    parser.add_argument("--topN", help="the maximum number of search hits to be printed", type=int)
+    parser.add_argument("--printIdsOnly", help="print only commentIds and not ids and their corresponding comments", action="store_true")
+    args = parser.parse_args()
+
+    # load queries
+    shouldBuildIndex = False
+    commentsFile, authorFile, articleFile = COMMENTSFILE, AUTHORFILE, ARTICLEFILE
+    if args.query.startswith("Index:"):
```

Comment:
This will be the first line in the queries textfile

Reply:
That's the way I understood the assignment the first time, as well. Therefore, I kept the possibility to have both (the first line in the query-file and as the name of the file)